File: problems/字符串/58.最后一个单词的长度.cpp
#include <iostream>
#include <string>
using namespace std;

int lengthOfLastWord(string s) {
    int n = s.size(), len = 0;
    for (int i = n - 1; i >= 0; i--) {
        if (s[i] != ' ')
            len++;
        else if (len > 0)
            break;
    }
    return len;
}

int main() {
    // 测试用例
    cout << lengthOfLastWord("Hello World") << endl;                    // 预期: 5
    cout << lengthOfLastWord("   fly me   to   the moon  ") << endl;  // 预期: 4
    cout << lengthOfLastWord("a") << endl;                             // 预期: 1
    return 0;
}

File: problems/字符串/58.最后一个单词的长度.py
#!/usr/bin/env python3
"""
58. 最后一个单词的长度
"""


def length_of_last_word(s: str) -> int:
    words = s.strip().split()
    return len(words[-1]) if words else 0


if __name__ == "__main__":
    # 测试用例
    print(length_of_last_word("Hello World"))      # 预期: 5
    print(length_of_last_word("   fly me   to   the moon  "))  # 预期: 4
    print(length_of_last_word("a"))               # 预期: 1
